### PR TITLE
Spooky October Bugfix

### LINKED ISSUE #436: == Ligature not working
# Environment

```none
Cascadia Code version number: 2102.25
Application (with version) used to display text: VSCode 1.54.3 and Windows Font Settings
OS platform and version: Windows 10 Pro 20H2
Screen resolution (i.e. 220dpi): 163 dpi

Any other software?
```

# Steps to reproduce

1. Install Cascadia Code
2. Type == in either vscode or the windows font preview

I tried this on a much older version of Cascadia code, version 1911.20 and it worked as expected. Does not work on the later versions.

# Expected behavior

1. Expect ligature for == to show

Version 2102.25

Version 1911.20

# Actual behavior

1. No ligature shows as seen in first image

Only happens for the == ligature as far as I know as ===, !=, etc all work fine.

### LINKED ISSUE #488: Some letter ligatures might wrong
# Environment
> Cascadia Code version number: 2102.025
# Some wrong ligatures
* Didn't support ligatures `|=` `==` but used to support. (Or for some reason I don't know.)
* `=>>` `=<<` ligatures didn't support in italic, but support in normal font-style.
* `www` ligatures in itatic seems has **7** letter "v"??? (Likes *vvvvvvv* .)

### LINKED ISSUE #521: Backtick could be more distinct from apostrophe
# Description of the new feature/enhancement

The backtick character is not easy to quickly distinguish from the apostrophe in code.
image

Is it possible to make it thicker and/or longer?

### LINKED ISSUE #524: Feature Request: Exclamation point clarity at small sizes
# Description of the new feature/enhancement (with images if possible)

In the exclamation point, at small sizes (pictured here at SemiLight size 12 in iTerm 2), the "dot" blends in too easily to the main stroke:

cascadia

For comparison, here's Menlo at the same size:

menlo

And IBM Plex Mono:

ibm plex mono

# Proposed technical implementation details (optional)

Increase space between dot and stroke, or give some shape to the stroke to better differentiate it from the dot.

### LINKED ISSUE #525: Feature Request: Clearer curly braces
# Description of the new feature/enhancement (with images if possible)

The shape of curly braces is not well-defined at small sizes and can resemble a parenthesis. Here it is in SemiLight 12 in iTerm 2:

braces-cascadia

For comparison, here's Menlo:

braces-menlo

And IBM Plex Mono:

braces-ibm-plex-mono

# Proposed technical implementation details (optional)

Make the shape of the curly brace more pronounced so that it's more legible at small sizes.

### LINKED ISSUE #529: Copyright says 2020 in 2021
### Cascadia family version

2106.17

### Cascadia family variant(s)

Cascadia Code (the version with ligatures), Cascadia Mono (the version without ligatures)

### Font file format(s)

.ttf (variable)

### Platform

Windows 10

### Other Software

_No response_

### What happened?

2021-07-13 19-04-28 ApplicationFrameHost

### LINKED ISSUE #532: Missing Arabic positional glyphs
### Cascadia family version

2106.17

### Cascadia family variant(s)

Cascadia Code (the version with ligatures), Cascadia Mono (the version without ligatures)

### Font file format(s)

.ttf (variable), .ttf (static), .otf (static)

### Platform

macOS 10.14.6

### Other Software

_No response_

### What happened?

Some Arabic letters are missing some positional glyphs:

* U+0649 ARABIC LETTER ALEF MAKSURA: missing 'init' and 'medi'
* U+066F ARABIC LETTER DOTLESS QAF: missing 'init' and 'medi'
* U+06A2 ARABIC LETTER FEH WITH DOT MOVED BELOW: missing 'init', 'medi', and 'fina'
* U+08AA ARABIC LETTER REH WITH LOOP: missing 'fina'
* U+08B2 ARABIC LETTER ZAIN WITH INVERTED V ABOVE: missing 'fina'

ىىـ␤ٯٯـ␤ڢڢڢ␤ـࢪ␤ـࢲ

### LINKED ISSUE #535: Wrong hamzas in lam–alef ligatures with U+0672, U+0673, and U+0675
### Cascadia family version

2106.17

### Cascadia family variant(s)

Cascadia Code (the version with ligatures), Cascadia Mono (the version without ligatures)

### Font file format(s)

.ttf (variable), .ttf (static), .otf (static)

### Platform

macOS 10.14.6

### Other Software

_No response_

### What happened?

In lam–alef ligatures, the hamzas of U+0672 ARABIC LETTER ALEF WITH WAVY HAMZA ABOVE, U+0673 ARABIC LETTER ALEF WITH WAVY HAMZA BELOW, and U+0675 ARABIC LETTER HIGH HAMZA ALEF are wrong. The glyphs look fine but are assigned to the wrong characters.
لٲلٳلٵ
Here is the same string in Scheherazade to clarify the correct rendering:
لٲلٳلٵ

### LINKED ISSUE #540: Wrong dots in U+06AE, U+0752, and U+0761
### Cascadia family version

2106.17

### Cascadia family variant(s)

Cascadia Code (the version with ligatures), Cascadia Mono (the version without ligatures)

### Font file format(s)

.ttf (variable), .ttf (static), .otf (static)

### Platform

macOS 10.14.6

### Other Software

_No response_

### What happened?

The triple dots of U+06AE ARABIC LETTER KAF WITH THREE DOTS BELOW, U+0752 ARABIC LETTER BEH WITH THREE DOTS POINTING UPWARDS BELOW, and U+0761 ARABIC LETTER FEH WITH THREE DOTS POINTING UPWARDS BELOW are pointing the wrong way.
ڮݒݡ
Here is the same string in Scheherazade to clarify the correct rendering:
ڮݒݡ

### LINKED ISSUE #541: U+FBC0 and U+FBBC should be spacing
### Cascadia family version

2106.17

### Cascadia family variant(s)

Cascadia Code (the version with ligatures), Cascadia Mono (the version without ligatures)

### Font file format(s)

.ttf (variable), .ttf (static), .otf (static)

### Platform

macOS 10.14.6

### Other Software

_No response_

### What happened?

U+FBC0 ARABIC SYMBOL SMALL TAH ABOVE and U+FBBC ARABIC SYMBOL DOUBLE VERTICAL BAR BELOW are spacing symbols, not combining marks, but in Cascadia Code, they combine with the previous Arabic glyph. For example, here is <U+0628, U+FBC0, U+0628, U+FBBC>:
ب﯀ب﮼
Here is the same string in Noto Sans Arabic to clarify the correct rendering:
ب﯀ب﮼

### LINKED ISSUE #542: Mispositioned U+0670 in “اللّٰه”
### Cascadia family version

2106.17

### Cascadia family variant(s)

Cascadia Code (the version with ligatures)

### Font file format(s)

.ttf (variable), .ttf (static), .otf (static)

### Platform

macOS 10.14.6

### Other Software

_No response_

### What happened?

The superscript alef in the word “اللّٰه” appears above the heh. It should be above the second lam.
اللّٰه

### LINKED ISSUE #546: ij_acute is a mark glyph
### Cascadia family version

2106.17

### Cascadia family variant(s)

Cascadia Code (the version with ligatures), Cascadia Mono (the version without ligatures)

### Font file format(s)

.ttf (variable), .ttf (static), .otf (static)

### Platform

macOS 10.14.6

### Other Software

_No response_

### What happened?

<U+0133 LATIN SMALL LIGATURE IJ, U+0301 COMBINING ACUTE ACCENT> overlaps the following glyph when the language system is Dutch.
ĳ́?
Intended rendering:
ĳ́?

### LINKED ISSUE #549: Wrong glyph for U+0609 ARABIC-INDIC PER MILLE SIGN
### Cascadia family version

2106.17

### Cascadia family variant(s)

Cascadia Code (the version with ligatures), Cascadia Mono (the version without ligatures)

### Font file format(s)

.ttf (variable), .ttf (static), .otf (static)

### Platform

macOS 10.14.6

### Other Software

_No response_

### What happened?

U+0609 ARABIC-INDIC PER MILLE SIGN looks like U+066A ARABIC PERCENT SIGN. U+0609 should have two dots at the bottom, not just one.

### LINKED ISSUE #555: Lam and alef presentation forms should not ligate
### Cascadia family version

2106.17

### Cascadia family variant(s)

Cascadia Code (the version with ligatures), Cascadia Mono (the version without ligatures)

### Font file format(s)

.ttf (variable), .ttf (static), .otf (static)

### Platform

macOS 10.14.6

### Other Software

_No response_

### What happened?

Arabic presentation forms represent specific glyphs that are immune to contextual shaping. When a lam presentation form precedes an alef presentation form, they should not be ligated. To get a lam–alef ligature using presentation forms, you should use a character like U+FEFB ARABIC LIGATURE LAM WITH ALEF ISOLATED FORM.

Here is <U+FEDF, U+FE8E, U+FEDF, U+FE84, U+FEDF, U+FE88, U+FEDF, U+FE82, U+FEDF, U+FB51, U+FEDF, U+FD3C, U+FEE0, U+FE8E, U+FEE0, U+FE84, U+FEE0, U+FE88, U+FEE0, U+FE82, U+FEE0, U+FB51, U+FEE0, U+FD3C>:
ﻟﺎﻟﺄﻟﺈﻟﺂﻟﭑﻟﴼﻠﺎﻠﺄﻠﺈﻠﺂﻠﭑﻠﴼ
It should look like this:
ﻟﺎﻟﺄﻟﺈﻟﺂﻟﭑﻟﴼﻠﺎﻠﺄﻠﺈﻠﺂﻠﭑﻠﴼ

### LINKED ISSUE #563: Wrong italic localised Serbian glyph U+0434
### Cascadia family version

2106.17

### Cascadia family variant(s)

Cascadia Code (the version with ligatures), Cascadia Mono (the version without ligatures)

### Font file format(s)

.ttf (static), .otf (static)

### Platform

Linux (freetype 2.10.4)

### Other Software

_No response_

### What happened?

Localised Serbian glyph U+0434 CYRILLIC SMALL LETTER DE should look like Bulgarian variation.

The image below shows Russian, Bulgarian and Serbian variants.
image

### LINKED ISSUE #571: Italic `i`'s, `j`'s and `r`'s stem top out of blue zone by 1 font unit
### Cascadia family version

2106.17

### Cascadia family variant(s)

Cascadia Code (the version with ligatures), Cascadia Mono (the version without ligatures)

### Font file format(s)

.otf (static)

### Platform

Linux (freetype 2.11.0), Windows 10

### Other Software

_No response_

### What happened?

The hinting for the Italic OTF's `i`, `j` and `r` is broken at various sizes. The top of the stem is hovering above the x-height blue zone by 1 font unit.

image psd

grafik

### LINKED ISSUE #572: Add support for the Unicode Ratio Character ( " ∶ " U+2236 )
# Ratio character on left, colon on right — ∶ :

At least in Windows 10, the Unicode ratio character is used instead of a colon when displaying the time on the taskbar. The ratio character is almost the same as the colon character, but it is slightly raised. Since the ratio character isn't supported in Cascadia Code, using it will display a box instead of the correct character. 

Time

Implementing the ratio character would be useful because it can be used between numbers like 3∶2 and subjectively looks better than 3:2 with a colon.

### LINKED ISSUE #577: Thin vertical line box drawing character U+2502 does not fill the line height
### Cascadia family version

2108.26

### Cascadia family variant(s)

Cascadia Mono (the version without ligatures)

### Font file format(s)

.ttf (variable)

### Platform

Windows 10

### Other Software

_No response_

### What happened?

Some of the box drawing characters, such as U+2588, the full cell solid block, fill up the entire vertical line space:
image

However, I can't find a thin, centered, vertical line that does the same. U+2502, the centered thin vertical line character from the same box drawing set, doesn't do so:
image

This makes terminal UI look pretty awful (for instance, TMUX vertical splits):
image

### PR DESCRIPTION
**Pull request by aaronbell**
## Summary of the Pull Request
This is a fairly comprehensive 🐛💀 update resolving many open issues.

## PR Checklist

1) Arabic bugfixes
- [x] Closes #532 👻 - Additional positional variants added
- [x] Closes #535 🍂 - Corrected hamza form
- [x] Closes #540 🎃 - Dot arrangement corrected
Screen Shot 2021-10-22 at 9 15 47 AM

- [x] Closes #541 🧹 - Was due to the use of anchors on those glyphs. These have been removed so the glyph can render as spacing.
- [x] Closes #542 🌕 - This was partly due to a [bug in Harfbuzz](https://github.com/harfbuzz/harfbuzz/issues/3069#issuecomment-888472442). It has been resolved both on the font side (through a different implementation) and in Harfbuzz. 
- [x] Closes #549 🦸‍♀️ - Design corrected
Screen Shot 2021-10-22 at 9 14 04 AM

- [x] Closes #555 💀 - All letter glyphs removed from Arabic Presentation form unicode slots to avoid situations where the glyphs are not behaving as expected.

Also:
#543 - uni0615 removed as Cascadia Arabic not intended to support Quranic

2) Other bug fixes
- [x] Closes #488 🔪 - Finally made the www ligature have the proper number of `w`s. 
Screen Shot 2021-10-22 at 9 13 02 AM

- [x] Closes #436 🧟‍♀️ - Extended length of Powerline 'caps' to avoid situations where rounding can prevent overlap. This may cause problems if the caps are used next to one another, but that seems an unlikely scenario given what I've reviewed of Powerline styles. 
Screen Shot 2021-10-22 at 9 17 34 AM

- [x] Closes #521 🤖 - enlarged the size of the grave character to make it more recognizable / legible in code. 
Screen Shot 2021-10-22 at 9 19 52 AM

- [x] Closes #524 ☠️ - Added some more differentiation in stroke, and also created more space using hinting. 
Screen Shot 2021-10-22 at 9 20 25 AM

- [x] Closes #525 🧙‍♂️ - tweaked the braces to be more twisty and create better differentiation from the parens. 
Screen Shot 2021-10-22 at 9 20 58 AM

- [x] Closes #529 🧛‍♀️ - Changed year :P
- [x] Closes #546 👹 - ij no longer masquerading as a mark. 
- [x] Closes #563 🧟‍♂️ - corrected `locl` feature for proper Serbian rendering
Screen Shot 2021-10-22 at 9 22 40 AM

- [x] Closes #571 🦹‍♀️ - corrected overshoot
- [x] Closes #572 🕷 - ratio symbol added
- [x] Closes #577 🍁 - shifted heights of box drawing lines to better align with block glyphs. Will reduce risk of non-joining forms under certain conditions.

### CODE REVIEW

File: sources/CascadiaCode-Bold.ufo/fontinfo.plist [hunk 1]
```diff
@@ -32,7 +32,7 @@
         <key>x</key>
         <integer>886</integer>
         <key>y</key>
-        <integer>-20</integer>
+        <integer>-1040</integer>
```
Comment (DHowett) [inline sources/CascadiaCode-Bold.ufo/fontinfo.plist:35]: these guideline changes aren't anything to worry about, right?

Comment (aaronbell) [reply sources/CascadiaCode-Bold.ufo/fontinfo.plist:35]: Nah. That's just a guideline. Won't affect anything.

**Comment by DHowett** [comment]
Woot! Thanks!

**Comment by aaronbell** [comment]
Seeing all those issues close at once is happy happy 😄 !

**Pull request merged**